packages/widget/src/stores/form/useFieldActions.ts

```diff
@@ -16,5 +56,43 @@ export const useFieldActions = () => {
     shallow,
   );
 
-  return actions;
+  const setFieldValueWithEmittedEvents = useCallback(
+    (
+      fieldName: FormFieldNames,
+      value: GenericFormValue,
+      options?: SetOptions,
+    ) => {
+      emitEventForFieldValueChange(
+        fieldName,
+        value,
+        actions.getFieldValues(fieldName)[0],
+        emitter,
+      );
+
+      actions.setFieldValue(fieldName, value, options);
```

Comment:
Should we set value first and, after that fire an event?

Reply:
The actions.getFieldValues will not have the old value to do the comparison to check for change if you call actions.setFieldValue first. I tried it the other way around initially